app/assets/javascripts/discourse/app/components/discovery/filter-navigation.gjs

```diff
@@ -81,6 +124,462 @@ export default class DiscoveryFilterNavigation extends Component {
     this._allowEnterSubmit = !value;
   }
 
+  @action
+  setupEventListeners() {
+    if (!this.inputElement) {
+      throw new Error("DiscoveryFilterNavigation requires an inputElement.");
+    }
+    this.inputElement.addEventListener("focus", this.handleInputFocus);
+    this.inputElement.addEventListener("blur", this.handleInputBlur);
+    this.inputElement.addEventListener("keydown", this.handleKeyDownTips);
+    this.inputElement.addEventListener("input", this.handleInput);
+  }
+
+  @action
+  onRegisterApi(api) {
+    this.dMenu = api;
+    if (this.inputElement) {
+      this.dMenu.trigger = this.inputElement;
+      this.dMenu.detachedTrigger = true;
+    }
+  }
+
+  @action
+  handleInput() {
+    this.updateResults(this.inputElement.value);
+  }
+
+  updateResults(value) {
+    this.newQueryString = value;
+    this.selectedIndex = -1;
+
+    const words = value.split(/\s+/);
+    const lastWord = words.at(-1);
+    const colonIndex = lastWord.indexOf(":");
+
+    if (colonIndex > 0) {
+      const prefix = this.extractPrefix(lastWord);
+      const filterName = lastWord.substring(
+        prefix.length,
+        colonIndex + prefix.length
+      );
+      const valueText = lastWord.substring(colonIndex + 1);
+
+      const tip = this.args.tips.find((t) => {
+        const tipFilterName = t.name.replace(/^[-=]/, "").split(":")[0];
+        return tipFilterName === filterName && t.type;
+      });
+
+      if (tip?.type) {
+        this.activeFilter = filterName;
+        this.handlePlaceholderSearch(filterName, valueText, tip, prefix);
+      } else {
+        this.activeFilter = null;
+        this.searchResults = [];
+      }
+    } else {
+      this.activeFilter = null;
+      this.searchResults = [];
+    }
+  }
+
+  @action
+  handleInputFocus() {
+    this.showTips = true;
+    this.selectedIndex = -1;
+    this.dMenu?.show();
+  }
+
+  @action
+  handleInputBlur() {
+    if (this.handleBlurTimer) {
+      cancel(this.handleBlurTimer);
+    }
```

Comment:
cancel will not generate an error on undefined values, so you don't need the if here